---
output: html_document
---

```{r, echo=FALSE}
library(knitr)
opts_chunk$set(fig.align = "center")
```

```{r, echo=TRUE, message=FALSE, warning=FALSE}
set.seed(1234567)
```

## `Seurat`

[Seurat](http://satijalab.org/seurat/) is a popular R package that is designed for QC, analysis, and exploration of single cell RNA-seq data, i.e. many of the tasks covered in this course. Although the authors provide several [tutorials](http://satijalab.org/seurat/get_started.html), here we provide a brief overview by following an [example](http://satijalab.org/seurat/pbmc3k_tutorial.html) created by the authors of `Seurat` (2,800 Peripheral Blood Mononuclear Cells). We mostly use default values in various function calls, for more details please consult the documentation and the authors. We start by loading the Deng data that we have used before:
```{r}
deng <- readRDS("deng/deng-reads.rds")
```

First, we initialize the Seurat object with the raw (non-normalized data). Keep all genes expressed in >= 3 cells. Keep all cells with at least 200 detected genes:
```{r, message=FALSE, warning=FALSE}
library(SingleCellExperiment)
library(Seurat)
seuset <- CreateSeuratObject(
    raw.data = counts(deng)
)
```

```{r, eval = FALSE}
VlnPlot(object = seuset, features.plot = c("nGene", "nUMI"), nCol = 3)

GenePlot(object = seuset, gene1 = "nUMI", gene2 = "nGene")

seuset <- FindVariableGenes(
    object = seuset
)

length(x = seuset@var.genes)

seuset <- ScaleData(object = seuset, vars.to.regress = c("nUMI"))

seuset <- RunPCA(
    object = seuset, 
    pc.genes = seuset@var.genes, 
    do.print = TRUE, 
    pcs.print = 1:5, 
    genes.print = 5
)

VizPCA(object = seuset, pcs.use = 1:2)

PCAPlot(object = seuset, dim.1 = 1, dim.2 = 2)

seuset <- ProjectPCA(object = seuset, do.print = FALSE)

PCHeatmap(
    object = seuset, 
    pc.use = 1:6, 
    cells.use = 500, 
    do.balanced = TRUE, 
    label.columns = FALSE,
    use.full = FALSE
)

seuset <- JackStraw(
    object = seuset, 
    num.replicate = 100, 
    do.print = FALSE
)

JackStrawPlot(object = seuset, PCs = 1:6)

PCElbowPlot(object = seuset)

seuset <- FindClusters(
    object = seuset, 
    reduction.type = "pca", 
    dims.use = 1:5, 
    resolution = 0.6, 
    print.output = 0, 
    save.SNN = TRUE
)

table(seuset@ident)

mclust::adjustedRandIndex(colData(deng)$cell_type2, seuset@ident)

# this did not work!
# Error in La.svd(x, nu, nv) : LAPACK routines cannot be loaded
# seuset <- RunTSNE(
#     object = seuset, 
#     dims.use = 1:5, 
#     do.fast = TRUE
# )
```

__Exercise 12__: Compare the results between `SC3` and `Seurat`.

__Our solution__:
```{r, echo=TRUE, eval=FALSE}
pData(deng)$Seurat <- as.character(seuset@ident)
sc3_plot_expression(deng, k = 10, show_pdata = "Seurat")
```


Seurat can also find marker genes, e.g. marker genes for cluster 2:
```{r, eval=FALSE}
markers <- FindMarkers(seuset, 2)
FeaturePlot(seuset, 
            head(rownames(markers)), 
            cols.use = c("lightgrey", "blue"), 
            nCol = 3)
```

__Exercise 13__: Compare marker genes provided by `Seurat` and `SC3`.

### sessionInfo()

```{r echo=FALSE}
sessionInfo()
```
